### DISCUSSION TITLE
"extends configuration" usage with remote repository from custom GitLab instance

### DISCUSSION THREAD

**Discussion by ColinEspinas** [nuxt/nuxt#16315] [Questions]
Hi,

We are currently considering using the `extends` configuration option to manage our design system with Nuxt3.

We managed to get it working with a local path (`extends: ['./ui']` like the [documention example](https://nuxt.com/docs/examples/advanced/config-extends)) but it seems impossible at the moment to use it with our custom GitLab instance remote repositories (`extends:['https://ourcustomgitlab.netatmo.com/web/ui']`).

When starting nuxt we hit a 404:
```
ERROR  Cannot restart nuxt:  {
  code: 404,
  message: 'Not Found'
}
```

The `gitlab:` option seems to only be usable in case of using gitlab.com.

Do you have any idea how this could be done? This feature is at the moment very poorly documented (would it be better to use custom nuxt modules instead ?).

**Comment by kewinzaq1** [comment]
@ColinEspinas  Did you handle it?

**Reply by ColinEspinas** [reply]
No, still no solution, we are considering using git submodules for the time being. It is not optimal but works.

**Comment by dsvgl** [comment]
Just blindly some things to check:

Did you try to create a dummy layer/repo on gitlab.com or github.com to see if that would work?
Also, there is a note in the [Git repository section](https://nuxt.com/docs/guide/going-further/layers#git-repository) that says you have to provide a token for private remote sources.

**Reply by ColinEspinas** [reply]
The `GIGET_AUTH` variable is used for private remote sources not for self-hosted instances, I believe.

**Reply by dsvgl** [reply]
ok, sounds like you should create a feature-request-issue. Or bug-report-issue :)
